src/computed fields.js

```diff
@@ -0,0 +1,161 @@
+/**
+🔊
+@file computed fields
+@summary Use JavaScript to dynamically calculate fields that are normally static.
+@license MIT
+@author Violgamba (Jon Heard)
+
+@description
+Use JavaScript to dynamically calculate fields that are normally static.  This allows for events
+that automatically change based on the game's state.  For example, an npc event's graphic field can
+be computed based on whether a "food" event is in the room.  If the player removes the "food" event,
+then the npc event's graphic automatically changes.
+
+This plugin is useful if the game has many state changes and/or many events reacting to state changes.
+
+The plugin field "computed-fields" lets you specify which fields can be computed.  See comments for
+this field at the bottom of this header.
+
+
+HOW TO USE - COMPUTE AVATAR'S GRAPHIC BASED ON WHICH SIDE OF THE ROOM THEY ARE ON
+1. Import this plugin into a game with an open room to move about in.
+2. Create two tiles for two avatar graphics.
+3. Create two tile fields on the Avatar event.  Name them "graphic-1" and "graphic-2".  Set them to the tiles of step 2.
+4. Change the avatar event's "graphic" field to be "javascript" type.  Set it to this code:
+  let f = "graphic-1";
+  if (EVENT.position[0] > 7) f = "graphic-2";
+  let data = FIELD(EVENT, f, "tile");
+  return { type: "tile", data: data };
+5. Run the game and move the player horizontally across the room.  Note that it's graphic changes halfway across.
+6. BONUS: Note that the avatar shows no graphic in the editor.  You can fix this by adding a "tile"
+		type field named "graphic" anywhere below the field of step 4.  The editor will use this
+		field for the graphic, while the playback will use the field of step 4 for the graphic.
+
+
+HOW TO USE - COMPUTE TERRAIN PASSABILITY BASED ON WHETHER AN ITEM IS TAKEN
+1. Import this plugin into a game with an open room to move about in.
+2. Draw a "boots" tile for an item.
+3. Draw a "spikes" tile for the floor.
+4. Add a "message" event beside the player.  Set the "say" field to "I put on the boots.".  Add a tag
+		field and name it "boots".  Add a tile field named "graphic" and set it to the boots tile of step 2.
+5. Add a "character" event a few cells from the avatar.  Set its graphic to the spikes tile of step 3.  Erase the "say" field.
+6. Change the "solid" tag field for the event of step 5 to javascript type.  Set it to this code:
+  if (!FIND_EVENT("boots")) return;
+  return { type: "tag", data: true };
+7. Copy the event of step 5.  Paste it around the player and the boots so that they are surrounded.
+8. Playtest the game.  Try to step over the spikes without getting the boots.  You should be blocked.  Walk over the boots and now the spikes should not block you.
```

Comment:
i would strongly recommend rewriting this example in terms of a locked door + key, which is what people tend to ask how to do